python/MaterialXTest/tests_to_html.py

```diff
@@ -4,37 +4,151 @@
 import os
 import datetime
 import argparse
+import json
 
+REDUCE_ENABLED = False
+DIFF_ENABLED = False
 try:
-    # Install pillow via pip to enable image differencing and statistics.
-    from PIL import Image, ImageChops, ImageStat
+    # Load modules for image diff, resizing and base64 encoding.
+    import base64
+    import cv2
+    import numpy as np
+    REDUCE_ENABLED = True
     DIFF_ENABLED = True
 except Exception:
-    DIFF_ENABLED = False
-
-def computeDiff(image1Path, image2Path, imageDiffPath):
-    try:
-        if os.path.exists(imageDiffPath):
-            os.remove(imageDiffPath)
-
-        if not os.path.exists(image1Path):
-            print ("Image diff input missing: " + image1Path)
-            return
-
-        if not os.path.exists(image2Path):
-            print ("Image diff input missing: " + image2Path)
-            return
-
-        image1 = Image.open(image1Path).convert('RGB')
-        image2 = Image.open(image2Path).convert('RGB')
-        diff = ImageChops.difference(image1, image2)
-        diff.save(imageDiffPath)
-        diffStat = ImageStat.Stat(diff)
-        return sum(diffStat.rms) / (3.0 * 255.0)
-    except Exception:
-        if os.path.exists(imageDiffPath):
-            os.remove(imageDiffPath)
-        print ("Failed to create image diff between: " + image1Path + ", " + image2Path)
+    print("- OpenCV and NumPy need to be installed for image diff and base64 for reduced image features to be supported.")
+
+class OpenCVImageUtils:
+    def __init__(self):
+        self.compute_reduced : bool = False
+        self.reduced_width : int = 512
+        self.difference_method = 'RMS'
+        self.MAX_DIFFERENCE : float = 1.0
+        self.hash_function = None
+        self._reset()
+
+    def _reset(self):
+        self.difference_image : str = ""
+        self.difference_value : float = self.MAX_DIFFERENCE
+        self.difference_image : str = ""
+
+    def set_reduced(self, compute_reduced: bool, width: int):
+        self.compute_reduced = compute_reduced
+        self.reduced_width = width
+        if self.reduced_width <= 16:
+            self.reduced_width = 16
+
+    def set_difference_method(self, method: str):
+        self.difference_method = method
+        if method == 'COLORMOMENT':
+            self.hash_function = cv2.img_hash.ColorMomentHash_create()
```

Comment:
We could add more hash comparisons here including ones which look for geometric details like `MarrHildrethHash`.